src/dialogs/more-info/controls/more-info-vacuum.ts

```diff
@@ -263,11 +263,8 @@ class MoreInfoVacuum extends LitElement {
       ? this.hass.states[batteryEntity.entity_id]
       : undefined;
 
-    const batteryIsBinary =
-      battery && computeStateDomain(battery) === "binary_sensor";
-
     // Use device battery entity
-    if (battery && (batteryIsBinary || !isNaN(battery.state as any))) {
+    if (battery && !isNaN(battery.state as any)) {
```

Comment:
Why this change?

Reply:
battery contains the result of findBatteryEntity, which is now capable of detecting the domain as suggested above by Piitaya. The check is no longer needed until a bit below where we need to know if the battery is a sensor. 
FindBatteryEntity now only returns sensors and binary sensors, not longer number entities since these are configurable.

Reply:
We must keep this condition because if it's a binary sensor, it display a battery with charging state.

Reply:
I see, thank you for the clarification. I've added the condition back in.